src/use-query.ts

```diff
@@ -73,6 +74,17 @@ export function useQuery<TResult, TError = ErrorDefault>(
   const refresh = () => store.refresh(entry.value)
   const refetch = () => store.refetch(entry.value)
 
+  // if passed by client
+  // & check if watchable
+  if (_options.enabled !== undefined
+      && (isRef(options.enabled) || typeof options.enabled === 'function')) {
```

Comment:
wouldn't checking the typeof to be boolean with options be enough? Since we have the defaults, it should be enough to skip adding a watcher for a primitive

Reply:
thank you, looks much better! I have added another commit